### PR TITLE
Test out first version of image tracking

### PR DESCRIPTION
**Pull request by HedwigAR**
Not done yet, but a first implementation of a web ar image tracking widget.

### CODE REVIEW

File: Web/Container/src/WebARContainer.tsx [hunk 1]
```diff
@@ -18,58 +31,75 @@ export function WebARContainer(props: WebARContainerContainerProps): ReactElemen
...
-        const canvas = newScene.getEngine().getRenderingCanvas();
+    useEffect(() => {
+        if (scene === undefined) {
+            const engine = new Engine(canvasRef.current, true, { xrCompatible: true }, true);
+            function updateSize() {
+                console.log("Resize");
+                engine.resize();
+            }
+
+            window.addEventListener("resize", updateSize);
```
Comment (gjulivan) [inline Web/Container/src/WebARContainer.tsx]: remove eventListener on unmount

File: Web/ImageTracker/src/WebARImageTracker.tsx [hunk 2]
```diff
@@ -73,70 +74,81 @@ export function WebARImageTracker(props: WebARImageTrackerContainerProps): React
     }, []);
 
     useEffect(() => {
-        if(engineContext.scene){
+        if(engineContext.scene && !observableSet){
```
Comment (iobuhov) [inline Web/ImageTracker/src/WebARImageTracker.tsx]: What is the main change here? Why we need this observableSet?

Comment (HedwigAR) [reply Web/ImageTracker/src/WebARImageTracker.tsx]: Oh this is to make sure this thing only happens once. It wasn't getting called or was getting called too often in a previous iteration.

File: Web/Container/src/WebARContainer.tsx (hunk 1 above)
Comment (HedwigAR) [reply Web/Container/src/WebARContainer.tsx]: Thanks, done

File: Web/ImageTracker/src/WebARImageTracker.tsx (hunk 2 above)
Comment (iobuhov) [reply Web/ImageTracker/src/WebARImageTracker.tsx]: As I understand main way to react on changes is `useEffect`, but is there any other ways to add "reactivity"?

**Review by gjulivan — APPROVED** [review]

**Comment by gjulivan** [comment]
general note: need more improvement on function readability and useCallback and useMemo,
due to it being the first version, i think it is fine.

**Pull request merged**